rule-syntax: Update behavior of metavariable-comparison strip

underyx opened the pull request:
Changed in https://github.com/returntocorp/semgrep/pull/4726

### Security

- [x] Change has no security implications (otherwise, ping the security team)

adamkvitek commented on docs/writing-rules/rule-syntax.md:
```diff
@@ -211,19 +211,22 @@ Comparison expressions support simple arithmetic as well as composition with [bo
 
 This will interpret metavariable values found in code as octal, so `0700` will be detected, but `0400` will not.
 
+Second, `strip: bool` makes Semgrep extract the first group of digits when a string is matched.
```
```suggestion
Second, when a string is matched, with `strip: bool`, Semgrep extracts the first group of digits.
```

adamkvitek commented on docs/writing-rules/rule-syntax.md:
```diff
@@ -211,19 +211,22 @@ Comparison expressions support simple arithmetic as well as composition with [bo
 
 <iframe src="https://semgrep.dev/embed/editor?snippet=AlqB" border="0" frameBorder="0" width="100%" height="435"></iframe>
 
 This will remove quotes (`'`, `"`, and `` ` ``) from both ends of the metavariable content. So `"2147483648"` will be detected but `"2147483646"` will not. This is useful when you expect strings to contain integer or float data.
```
```suggestion
This removes quotes (`'`, `"`, and `` ` ``) from both ends of the metavariable content. So `"2147483648"` is detected, but `"2147483646"` not. This is useful when you expect strings to contain integer or float data.
```

adamkvitek replied:
Present simple suffices. :-)

adamkvitek commented on docs/writing-rules/rule-syntax.md:
```diff
@@ -211,19 +211,22 @@ Comparison expressions support simple arithmetic as well as composition with [bo
 <iframe src="https://semgrep.dev/embed/editor?snippet=AlqB" border="0" frameBorder="0" width="100%" height="435"></iframe>
 
 This will remove quotes (`'`, `"`, and `` ` ``) from both ends of the metavariable content. So `"2147483648"` will be detected but `"2147483646"` will not. This is useful when you expect strings to contain integer or float data.
+It will also remove any other non-digit characters. This lets you match a string like `"12px"` and compare against the number `12` from within that string.
```
```suggestion
It also removes any other non-digit characters. This enables you to match a string like `"12px"` and compare it against the number `12`.
```

adamkvitek replied:
I like that you speak directly to the audience by using the second person!

adamkvitek reviewed (commented):
@underyx thank you for your addition, I made a few comments, feel free to check them out.

underyx commented on docs/writing-rules/rule-syntax.md:
```diff
@@ -205,25 +205,38 @@ The `metavariable-comparison` operator is a mapping which requires the `metavari
+- arithmetic operators: `+`, `-`, `*`, `/`, `%`
+- comparison operators: `>`, `>=`, `==`, `!=`, `<=`, `<`, `in`
+- boolean logic operators: `not`, `and`, `or`
+- the Python `int(value)` function
```
```suggestion
- the Python `int(value: Any) -> str` function
```

underyx commented on docs/writing-rules/rule-syntax.md:
```diff
@@ -205,25 +205,38 @@ The `metavariable-comparison` operator is a mapping which requires the `metavari
+- arithmetic operators: `+`, `-`, `*`, `/`, `%`
+- comparison operators: `>`, `>=`, `==`, `!=`, `<=`, `<`, `in`
+- boolean logic operators: `not`, `and`, `or`
+- the Python `int(value: Any) -> str` function
```
```suggestion
- the Python `int(value: Any) -> int` function
```

underyx commented:
Thanks @adamkvitek! I incorporated the changes you requested, brought in updates to the logic based on the original semgrep PR's review, and added some more content:

https://github.com/returntocorp/semgrep-docs/pull/453/files#diff-27ce63caf629155835158f784d2556151d037d7018d5618da4eab9cce3195162R208-R214

adamkvitek commented:
@underyx just small review comments. Looks fine, if this feature is out and working, feel free to implement these tiny suggestions and I will approve it for you. :-)

adamkvitek commented:
@underyx just looking at this PR, are we going to resurrect it? :-) (Sorry for poking, I finally found a time to check all our PRs in this repository.)

underyx commented:
> @underyx just looking at this PR, are we going to resurrect it? :-) (Sorry for poking, I finally found a time to check all our PRs in this repository.)

not likely within the next two weeks!
